### ISSUE TITLE
Passing array of styles causes PHP Warnings and Notices

### ISSUE THREAD

**Issue by sanzeeb3** [WordPress/gutenberg#43086]
### Description

In the documentation: https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#style

> Note: An option to pass also an array of styles exists since WordPress 5.9.0.

passing an array of styles is possible since 5.9.0. However, I'm on WordPress 6.1 and there are some errors and notices:

> [09-Aug-2022 11:04:58 UTC] PHP Notice:  Array to string conversion in C:\laragon\www\dev\wp-includes\block-editor.php on line 340
> [09-Aug-2022 11:04:58 UTC] PHP Warning:  explode() expects parameter 2 to be string, array given in C:\laragon\www\dev\wp-includes\class.wp-dependencies.php on line 186
> [09-Aug-2022 11:04:58 UTC] PHP Notice:  Trying to access array offset on value of type null in C:\laragon\www\dev\wp-includes\class.wp-dependencies.php on line 187

Both the styles are enqueued, though.

### Step-by-step reproduction instructions

1) Register a block with an array of styles. E.g.

```
register_block_type(
	'wpfda/block',
       [
                'style' => [ 'wpfda-style', 'wpfda-editor-style' ]
       ]
```

2) Go to new post page.
3) See the error.

### Environment info

Gutenberg is not installed.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by gziolo** [comment]
Passing an array of styles works when passed through `block.json` file. It doesn't work today when using the 2nd param from `register_block_type`. There is a proposal to address that: https://github.com/WordPress/wordpress-develop/pull/2853. I hope to see it included in WordPress 6.1.

**Comment by gziolo** [comment]
I have an alternative proposal that should resolve the issue in WordPress core for all style and script types: https://github.com/WordPress/wordpress-develop/pull/3108.

**Comment by t-hamano** [comment]
A similar problem was reported in #44687. And these issues have been fixed in [Changeset 54670](https://core.trac.wordpress.org/changeset/54670) and appear to be included in WordPress 6.1.

Therefore, this issue may be closed.
